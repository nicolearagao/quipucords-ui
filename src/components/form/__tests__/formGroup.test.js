import React from 'react';
import { FormGroup } from '../formGroup';
import { formHelpers } from '../formHelpers';

describe('FormGroup Component', () => {
  it('should render a basic component', () => {
    const props = { id: 'test' };

    const component = renderComponent(
      <FormGroup label="test" {...props}>
        <input id="test" type="text" value="" readOnly />
      </FormGroup>
    );

    expect(component).toMatchSnapshot('basic formgroup');
    expect(component.find('label')).toMatchSnapshot('basic label');
  });

  it('should handle multiple error message types', () => {
    const props = {
      id: 'test',
      error: true,
      errorMessage: 'lorem ipsum'
    };

    const component = renderComponent(
      <FormGroup {...props}>
        <input id="test" type="text" value="" readOnly />
      </FormGroup>
    );

    expect(component).toMatchSnapshot('string error message');

    const componentErrorMessage = component.setProps({ errorMessage: true });
    expect(componentErrorMessage).toMatchSnapshot('boolean error message');

    const componentNodeErrorMessage = component.setProps({ errorMessage: <span>lorem ipsum</span> });
    expect(componentNodeErrorMessage).toMatchSnapshot('node error message');
  });

  it('should have isEmpty validation', () => {
    expect(formHelpers.isEmpty(undefined)).toBe(true);
    expect(formHelpers.isEmpty(null)).toBe(true);
    expect(formHelpers.isEmpty('')).toBe(true);
    expect(formHelpers.isEmpty('lorem')).toBe(false);
    expect(formHelpers.isEmpty([])).toBe(true);
    expect(formHelpers.isEmpty(['lorem'])).toBe(false);
    expect(formHelpers.isEmpty({})).toBe(true);
    expect(formHelpers.isEmpty({ lorem: 'ipsum' })).toBe(false);
  });

  it('should have doesNotHaveMinimumCharacters validation', () => {
    expect(formHelpers.doesNotHaveMinimumCharacters('', 5)).toBe(true);
    expect(formHelpers.doesNotHaveMinimumCharacters('test test', 5)).toBe(false);
  });

  it('should have isPortValid validation', () => {
    expect(formHelpers.isPortValid('lorem')).toBe(false);
    expect(formHelpers.isPortValid(-1)).toBe(false);
    expect(formHelpers.isPortValid(65536)).toBe(false);
    expect(formHelpers.isPortValid('65536')).toBe(false);
    expect(formHelpers.isPortValid(65535)).toBe(true);
  });
});
