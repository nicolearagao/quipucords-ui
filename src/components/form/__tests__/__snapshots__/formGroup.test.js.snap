// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup Component should handle multiple error message types: boolean error message 1`] = `
<formgroup
  props="{
  "id": "test",
  "error": true,
  "errorMessage": true,
  "children": [
    {
      "props": {
        "id": "test",
        "type": "text",
        "value": "",
        "readOnly": true
      },
      "type": "input"
    }
  ],
  "helperText": null,
  "fieldId": null
}"
>
  <div
    class="pf-c-form__group"
  >
    <div
      class="pf-c-form__group-control"
    >
      <input
        id="test"
        readonly=""
        type="text"
        value=""
      />
      <div
        aria-live="polite"
        class="pf-c-form__helper-text pf-m-error"
        id="test-helper"
      >
        Error
      </div>
    </div>
  </div>
</formgroup>
`;

exports[`FormGroup Component should handle multiple error message types: node error message 1`] = `
<formgroup
  props="{
  "id": "test",
  "error": true,
  "errorMessage": {
    "type": "span",
    "key": null,
    "ref": null,
    "props": {
      "children": "lorem ipsum"
    },
    "_owner": null,
    "_store": {}
  },
  "children": [
    {
      "props": {
        "id": "test",
        "type": "text",
        "value": "",
        "readOnly": true
      },
      "type": "input"
    }
  ],
  "helperText": null,
  "fieldId": null
}"
>
  <div
    class="pf-c-form__group"
  >
    <div
      class="pf-c-form__group-control"
    >
      <input
        id="test"
        readonly=""
        type="text"
        value=""
      />
      <span>
        lorem ipsum
      </span>
    </div>
  </div>
</formgroup>
`;

exports[`FormGroup Component should handle multiple error message types: string error message 1`] = `
<formgroup
  props="{
  "id": "test",
  "error": true,
  "errorMessage": "lorem ipsum",
  "children": [
    {
      "props": {
        "id": "test",
        "type": "text",
        "value": "",
        "readOnly": true
      },
      "type": "input"
    }
  ],
  "helperText": null,
  "fieldId": null
}"
>
  <div
    class="pf-c-form__group"
  >
    <div
      class="pf-c-form__group-control"
    >
      <input
        id="test"
        readonly=""
        type="text"
        value=""
      />
      <div
        aria-live="polite"
        class="pf-c-form__helper-text pf-m-error"
        id="test-helper"
      >
        lorem ipsum
      </div>
    </div>
  </div>
</formgroup>
`;

exports[`FormGroup Component should render a basic component: basic formgroup 1`] = `
<formgroup
  props="{
  "label": "test",
  "id": "test",
  "children": [
    {
      "props": {
        "id": "test",
        "type": "text",
        "value": "",
        "readOnly": true
      },
      "type": "input"
    }
  ],
  "error": null,
  "errorMessage": null,
  "helperText": null,
  "fieldId": null
}"
>
  <div
    class="pf-c-form__group"
  >
    <div
      class="pf-c-form__group-label"
    >
      <label
        class="pf-c-form__label"
        for="test"
      >
        <span
          class="pf-c-form__label-text"
        >
          test
        </span>
      </label>
       
    </div>
    <div
      class="pf-c-form__group-control"
    >
      <input
        id="test"
        readonly=""
        type="text"
        value=""
      />
      
    </div>
  </div>
</formgroup>
`;

exports[`FormGroup Component should render a basic component: basic label 1`] = `
<label
  class="pf-c-form__label"
  for="test"
>
  <span
    class="pf-c-form__label-text"
  >
    test
  </span>
</label>
`;
